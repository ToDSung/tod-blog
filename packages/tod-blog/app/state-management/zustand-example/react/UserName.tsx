import UserNameTemplate from '../../component/UserNameTemplate';
import useUserStore from '../store/userStore';

const UserName = () => {
  const {
    user: { name },
  } = useUserStore();

  return (
    <UserNameTemplate
      name={name}
      syntax='const { user: { name } } = useUserStore();'
    />
  );
};

export default UserName;
