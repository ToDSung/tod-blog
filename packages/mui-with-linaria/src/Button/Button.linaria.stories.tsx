import type { Meta, StoryObj } from '@storybook/react';
import { fireEvent, userEvent, within, fn, expect } from '@storybook/test';
import { ComponentProps, CSSProperties } from 'react';

import ButtonLinaria from './Button.linaria';

type ButtonPropsAndCustomArgs = ComponentProps<typeof ButtonLinaria> & {
  style?: CSSProperties;
};

const meta: Meta<ButtonPropsAndCustomArgs> = {
  title: 'mui-with-linaria/ButtonLinaria',
  component: ButtonLinaria,
  args: {
    label: 'Button Label',
    variant: 'contained',
    children: 'Button',
    onClick: fn(),
    style: { width: 100, height: 20, padding: 16 },
  },
  render: ({ style, ...props }) => (
    <div style={style}>
      <ButtonLinaria {...props} />
    </div>
  ),
} satisfies Meta<ButtonPropsAndCustomArgs>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {},
  play: async ({ canvasElement, args }) => {
    const canvas = within(canvasElement);

    expect(canvas.getByText('Button Label')).toHaveTextContent('Button Label');

    expect(canvas.getByLabelText('Button Label')).toHaveTextContent('Button');
  },
};

export const ClickExample: Story = {
  play: async ({ canvasElement, args }) => {
    const canvas = within(canvasElement);

    await userEvent.click(canvas.getByRole('button'));

    expect(args.onClick).toHaveBeenCalledTimes(1);

    await fireEvent.click(canvas.getByRole('button'));

    expect(args.onClick).toHaveBeenCalledTimes(2);
  },
};
