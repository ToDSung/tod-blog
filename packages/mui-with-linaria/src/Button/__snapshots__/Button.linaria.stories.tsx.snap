// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mui-with-linaria/ButtonLinaria ClickExample play-test 1`] = `
<div style="width: 100px; height: 20px; padding: 16px;">
  <div class="b1o0vl8n">
    <label id=":r1:">
      Button Label
    </label>
    <button class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary Mui-focusVisible MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary bc2jvtj css-sghohy-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
            aria-labelledby=":r1:"
    >
      Button
      <span class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root">
        <span class="css-y4cjyz-MuiTouchRipple-ripple MuiTouchRipple-ripple MuiTouchRipple-rippleVisible MuiTouchRipple-ripplePulsate"
              style="width: 84.3253px; height: 84.3253px; top: -24.1626px; left: 7.83737px;"
        >
          <span class="MuiTouchRipple-child MuiTouchRipple-childPulsate">
          </span>
        </span>
        <span class="css-y4cjyz-MuiTouchRipple-ripple MuiTouchRipple-ripple MuiTouchRipple-rippleVisible"
              style="width: 109.563px; height: 109.563px; top: -36.7814px; left: -4.78138px;"
        >
          <span class="MuiTouchRipple-child MuiTouchRipple-childLeaving">
          </span>
        </span>
      </span>
    </button>
  </div>
</div>
`;

exports[`mui-with-linaria/ButtonLinaria Default play-test 1`] = `
<div style="width: 100px; height: 20px; padding: 16px;">
  <div class="b1o0vl8n">
    <label id=":r0:">
      Button Label
    </label>
    <button class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary bc2jvtj css-sghohy-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
            aria-labelledby=":r0:"
    >
      Button
      <span class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root">
      </span>
    </button>
  </div>
</div>
`;
